lib/bolt/applicator.rb

```diff
@@ -162,7 +162,13 @@ def compile(target, scope)
 
     def validate_hiera_config(hiera_config)
       if File.exist?(File.path(hiera_config))
-        data = File.open(File.path(hiera_config), "r:UTF-8") { |f| YAML.safe_load(f.read, [Symbol]) }
+        data = File.open(File.path(hiera_config), "r:UTF-8") do |f|
+          if Psych.method(:safe_load).parameters.flatten.include?(:permitted_classes)
```

Comment:
I think instead of `flatten.include?` we can just do `[...].parameters.rassoc(:permitted_classes)`